src/components/ReportActionItem/ReportPreview.js

```diff
@@ -131,10 +135,16 @@ function ReportPreview(props) {
     const {getLineHeightStyle} = useStyleUtils();
     const {translate} = useLocalize();
 
-    const [hasMissingSmartscanFields, sethasMissingSmartscanFields] = useState(false);
-    const [areAllRequestsBeingSmartScanned, setAreAllRequestsBeingSmartScanned] = useState(false);
-    const [hasOnlyDistanceRequests, setHasOnlyDistanceRequests] = useState(false);
-    const [hasNonReimbursableTransactions, setHasNonReimbursableTransactions] = useState(false);
+    const {hasMissingSmartscanFields, areAllRequestsBeingSmartScanned, hasOnlyDistanceRequests, hasNonReimbursableTransactions} = useMemo(
+        () => ({
+            hasMissingSmartscanFields: ReportUtils.hasMissingSmartscanFields(props.iouReportID),
+            areAllRequestsBeingSmartScanned: ReportUtils.areAllRequestsBeingSmartScanned(props.iouReportID, props.action),
+            hasOnlyDistanceRequests: ReportUtils.hasOnlyDistanceRequestTransactions(props.iouReportID),
+            hasNonReimbursableTransactions: ReportUtils.hasNonReimbursableTransactions(props.iouReportID),
+        }),
+        // eslint-disable-next-line react-hooks/exhaustive-deps -- we just want this to run when transactions get updated
+        [props.transactions, props.iouReportID, props.action],
```

Comment:
So if we only want to trigger this function when transactions are updated. Is there another way to get the latest iouReportId and action? 

Maybe I'm wrong, but it looks like with the above code we'd trigger it when any of these three props changes, right?

Reply:
Yeah, I will write a better comment.

You are correct, but the eslint-disable is there only because of the `transactions` props, since we don't use the transactions inside the `useMemo` block it complains.

Reply:
Ah okay. It sounds like this is a common issue then, but just to check -- is there another hook type that lets us use props without needing to subscribe to changes?

Reply:
As far as I know you can use every prop inside the hook without subscribing to it, but lint will complain and I think it is usually not a good practice.

Reply:
Cool. As long as we're doing this elsewhere in the app already is seems okay 👍